src/action.ts

```diff
@@ -0,0 +1,81 @@
+import { Link } from "./link";
+import { Device } from "./device";
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+export interface IAction {
+    title: string;
+    type: string;
+    '@type': string;
+    description: string;
+    readOnly: boolean;
+    links: Link[];
+}
+
+export class Action {
+    private name_ : string;
+    private action : IAction;
+    private device_ : Device;
+    constructor(name_: string, action: IAction, device_: Device) {
+        this.name_ = name_;
+        this.action = action;
+        this.device_ = device_;
+    }
+    public get title() {
+        return this.action.title;
+    }
+    public get type() {
+        return this.action.type;
+    }
+    public get '@type'() {
+        return this.action['@type'];
+    }
+    public get description() {
+        return this.action.description;
+    }
+    public get readOnly() {
+        return this.action.readOnly;
+    }
+    public get links() {
+        return this.action.links;
+    }
+    public get name() {
+        return this.name_;
+    }
+    public get device() {
+        return this.device_;
+    }
+    public serialize() {
+        return this.action;
+    }
+    public async execute(input = {}) {
+        await this.device.client.post(this.actionHref, {[this.name]: {input: input}});
+    }
+    private get actionHref(): string {
```

Comment:
I wouldn't expect an expression like `this.actionHref` to throw an error.
It would be more apparent that this might fail if it would be a regular method.